colossus-metrics/src/main/scala/colossus/metrics/Aggregation.scala

```diff
@@ -38,20 +38,23 @@ object TagSelector {
 }
 
 sealed trait AggregationType {
-  def aggregate(values: Seq[Long]): Long
+  def aggregate(values: Seq[MetricValue]): RawMetricValue
 }
 object AggregationType {
   case object Sum extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum
+    def aggregate(values: Seq[MetricValue]): RawMetricValue = values.foldLeft(0L){case (build: RawMetricValue, next: MetricValue) => build + next.value}
   }
   case object Average extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.sum / values.size
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (build, next) => build + next.value} / values.size
   }
   case object Max extends AggregationType {
-    def aggregate(values: Seq[Long]) = values.max
+    def aggregate(values: Seq[MetricValue]) = values.foldLeft(0L){case (max, next) => if (max > next.value) max else next.value}
```

Comment:
Actually now that I'm thinking about, the way the aggregation works these will never get called if the lists are empty.  Basically aggregation happens while looping over value maps (`Map[TagMap, MetricValue]`) so either the seq will be non-empty or non-existent.  I'll still fix the issue for negative values though